tests/app/client/test_gf_api_client.py

```diff
@@ -79,3 +80,29 @@ def test_get_voronoi_analysis_raises_error():
 def test_get_voronoi_analysis_raises_error_for_none():
     with pytest.raises(ValueError, match=client.MISSING_UUID_ERROR):
         client.get_voronoi_analysis(None)  # type: ignore[arg-type]
+
+
+# Tests for the get_analysis_status function
+def test_get_analysis_status_succeeded():
+    response = client.get_analysis_status(client.SAMPLE_UUID)
+    assert response == AnalysisStatus.ANALYSIS_SUCCEEDED.value
+
+
+def test_get_analysis_status_in_progress():
+    response = client.get_analysis_status(client.IN_PROGRESS_UUID)
+    assert response == AnalysisStatus.ANALYSIS_IN_PROGRESS.value
+
+
+def test_get_analysis_status_failed():
+    response = client.get_analysis_status(client.ANALYSIS_FAILED_UUID)
+    assert response == AnalysisStatus.ANALYSIS_FAILED.value
+
+
+def test_get_analysis_status_not_found():
+    with pytest.raises(FileNotFoundError):
+        client.get_analysis_status("unknown-uuid")
+
+
+def test_get_analysis_status_no_uuid_provided():
+    with pytest.raises(ValueError, match=client.MISSING_UUID_ERROR):
+        client.get_analysis_status(None)
```

Comment:
If changing the `get_analysis_status` parameters to not accept `None` is getting you errors on this line, you can get the linter to ignore that by adding this comment here:

```suggestion
        client.get_analysis_status(None) # type: ignore[arg-type]
```